File: lucafontanili.designpatterns/src/main/java/com/lucafontanili/designpatterns/behavioral/memento/MementoClient.java
package com.lucafontanili.designpatterns.behavioral.memento;

import java.io.IOException;

import com.lucafontanili.designpatterns.utilities.AbstractClient;
import com.lucafontanili.designpatterns.utilities.InvalidArgumentException;

public class MementoClient extends AbstractClient {

    public static void main(String[] args) throws InvalidArgumentException, IOException {
	main(new MementoClient());
    }

    @Override
    public void run() throws InvalidArgumentException, IOException {
	Originator originator = new Originator();
	originator.setState("First");
	LOGGER.info(originator.getState());
	originator.setState("Second");
	LOGGER.info(originator.getState());
	LOGGER.info(originator.getHistory());
	LOGGER.info(new StringBuilder(8).append("Undo"));
	originator.undo();
	LOGGER.info(originator.getState());
	originator.setState("Third");
	LOGGER.info(originator.getState());

	LOGGER.info(originator.getHistory());
    }
}

File: lucafontanili.designpatterns.python/src/behavioral/memento/MementoClient.py
'''
Created on Nov 13, 2016

@author: Admin
'''

from Memento import *

def main():
    originator = Originator()
    originator.set_state("First")
    print(originator.state())
    originator.set_state('Second')
    print(originator.state())
    for memento in originator.hisotry():
        print memento.to_string()
    print('Undo')
    originator.undo()
    print(originator.state())
    originator.set_state('Third')
    print(originator.state())
    for memento in originator.hisotry():
        print memento.to_string()

if __name__ == "__main__":
    main()